proposals/field-keyword.md

```diff
@@ -230,74 +230,110 @@ public struct S
 
 ### Nullability
 
-When `{ get; }` is written as `{ get => field; }`, or `{ get; set; }` is written as `{ get => field; set => field = value; }`, a similar warning should be produced when a non-nullable property is not initialized:
+A principle of the the Nullable Reference Types feature was to understand existing idiomatic coding patterns in C# and to require as little ceremony as possible around those patterns. The `field` keyword proposal enables simple, idiomatic patterns to address widely asked-for scenarios, such as lazily initialized properties. It's important for the Nullable Reference Types to mesh well with these new coding patterns.
 
-```cs
-class C
-{
-    // ⚠️ CS8618: Non-nullable property 'P' must contain a
-    // non-null value when exiting constructor.
-    public string P { get => field; set => field = value; }
-}
-```
+Goals:
 
-No warning should be produced if the property is initialized to a non-null value via constructor assignment or property initializer:
+- A reasonable level of null-safety should be ensured for various usage patterns of the `field` keyword feature.
 
-```cs
-class C
-{
-    public C() { P = ""; }
+- Patterns that use the `field` keyword should feel as though they've always been part of the language. Avoid making the user jump through hoops to enable Nullable Reference Types in code that is perfectly idiomatic for the `field` keyword feature.
 
-    public string P { get => field; set => field = value; }
-}
-```
+One of the key scenarios is lazily initialized properties:
```

Comment:
I rewrote this goals/motivation section to include some of the rationale that came out of LDM, such as the thought experiment of designing the NRTs feature after the `field` keyword had already been in use.

Suggestions/nits welcome!